### PR TITLE
Bjorncs/reindexing cluster controller in config model

### PR DESCRIPTION
**Pull request by bjorncs**

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ReindexingController.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    private final String contentClusterName;
+    private final Collection<NewDocumentType> documentTypes;
+
+    ReindexingController(ReindexingContext context) {
+        super(new ComponentModel(
+                BundleInstantiationSpecification.getFromStrings(
+                        "reindexing-maintainer",
+                        "ai.vespa.reindexing.ReindexingMaintainer",
+                        REINDEXING_CONTROLLER_BUNDLE)));
+        this.reindexing = context.reindexing().orElse(null);
```
Comment (jonmv) [inline config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ReindexingController.java:30]: Can use the default interface here?

File: config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ReindexingController.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        this.contentClusterName = context.contentClusterName();
+        this.documentTypes = context.documentTypes();
+    }
+
+    @Override
+    public void getConfig(ReindexingConfig.Builder builder) {
+        builder.clusterName(contentClusterName);
+        if (reindexing == null || !reindexing.enabled()) {
```
Comment (jonmv) [inline config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ReindexingController.java:38]: That eliminates the need for nullity checks.

File: config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ReindexingController.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            builder.enabled(false);
+            return;
```
Comment (jonmv) [inline config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ReindexingController.java:40]: And we _can_ send the timestamps for all documents when disabled, as the thing won't run anyway. 

I think we should probably settle on using either signal for disabling it.

**Review by jonmv — APPROVED** [review]

**Comment by jonmv** [comment]
Let's consider using a `new Reindexing() { }` in place of `null`, and always sending the timestamps for each document type, since the `enabled bool` guards the actual reindexing anyway.

**Comment by bjorncs** [comment]
@jonmv PTAL

**Review by jonmv — APPROVED** [review]
Unicorn

**Pull request merged**